x-pack/plugin/sql/src/main/java/org/elasticsearch/xpack/sql/plugin/TransportSqlQueryAction.java

```diff
@@ -80,13 +83,14 @@ public static void operation(PlanExecutor planExecutor, SqlQueryRequest request,
             planExecutor.sql(cfg, request.query(), request.params(),
                     wrap(p -> listener.onResponse(createResponseWithSchema(request, p)), listener::onFailure));
         } else {
-            planExecutor.nextPage(cfg, Cursors.decodeFromString(request.cursor()),
-                    wrap(p -> listener.onResponse(createResponse(request, null, p)),
+            Tuple<Cursor, ZoneId> decoded = Cursors.decodeFromStringWithZone(request.cursor());
```

Comment:
Is `Cursors.decodeFromString` used anywhere else ? It might be that decoding a tuple of timezone or adding it as a property in the `Cursor` class should be the default.

Reply:
I'm planning to do that in an upcoming PR.

Reply:
Ad discussed, will not pass the zoneId into the Cursor, as zoneId is the responsibility of SqlInput/OutputStreams to handle. Instead, I remove the `decodeFromString` and only use the `decodeFromStringWithZone`, so it's obvious for consumers that the zoneId is always decoded as well and returned as part of the returned `Tuple`.